src/utils/filetype.rs

```diff
@@ -0,0 +1,70 @@
+const IMAGE_TYPES: [&str; 6] = ["jpg", "jpeg", "gif", "png", "svg", "webp"];
+const VIDEO_TYPES: [&str; 2] = ["mp4", "webm"];
+
+pub fn is_image(url: &str) -> bool {
+  is_filetype(url, &IMAGE_TYPES)
+}
+
+pub fn is_video(url: &str) -> bool {
+  is_filetype(url, &VIDEO_TYPES)
+}
+
+fn is_filetype(url: &str, exts: &[&str]) -> bool {
```

Comment:
Something tells me that using `path_segments.last()` from https://docs.rs/url/latest/url/ might be a little safer, but you've added enough tests here that this is pry fine.

Reply:
Creating a `Url` with the `url` crate allocates the entire url as a string on the heap (as would calling `to_lowerrcase` on the `url` param), whereas this approach only allocates a `String` for the file extension of the url. I'm arguably being too anal about performance here, but I personally don't find the code too difficult to understand, its use is isolated to only this function as the single source of truth, and, as you pointed out, it has good test coverage.

Reply:
Oh yeah I forgot about that... yep this is fine then.